File: seqkit/replace.go
package seqkit

import "ignis/driver/api"

type SeqKitReplaceOptions struct {
	inner ReplaceOptions
}

type ReplaceOptions struct {
	Config      KitConfig
	Pattern     *string
	Replacement *string
	NrWidth     *int
	BySeq       *bool
	IgnoreCase  *bool
	KvFile      *string
	KeepUntouch *bool
	KeepKey     *bool
	KeyCaptIdx  *int
	KeyMissRepl *string
}

func (this *ReplaceOptions) setDefaults() *ReplaceOptions {
	this.Config.setDefaults()
	setDefault(&this.Pattern, "")
	setDefault(&this.Replacement, "")
	setDefault(&this.NrWidth, 1)
	setDefault(&this.BySeq, false)
	setDefault(&this.IgnoreCase, false)
	setDefault(&this.KvFile, "")
	setDefault(&this.KeepUntouch, false)
	setDefault(&this.KeepKey, false)
	setDefault(&this.KeyCaptIdx, 1)
	setDefault(&this.KeyMissRepl, "")

	return this
}

func (this *SeqKitReplaceOptions) Config(v *SeqKitConfig) *SeqKitReplaceOptions {
	this.inner.Config = v.inner
	return this
}

func (this *SeqKitReplaceOptions) Pattern(v string) *SeqKitReplaceOptions {
	this.inner.Pattern = &v
	return this
}

func (this *SeqKitReplaceOptions) Replacement(v string) *SeqKitReplaceOptions {
	this.inner.Replacement = &v
	return this
}

func (this *SeqKitReplaceOptions) NrWidth(v int) *SeqKitReplaceOptions {
	this.inner.NrWidth = &v
	return this
}

func (this *SeqKitReplaceOptions) BySeq(v bool) *SeqKitReplaceOptions {
	this.inner.BySeq = &v
	return this
}

func (this *SeqKitReplaceOptions) IgnoreCase(v bool) *SeqKitReplaceOptions {
	this.inner.IgnoreCase = &v
	return this
}

func (this *SeqKitReplaceOptions) KvFile(v string) *SeqKitReplaceOptions {
	this.inner.KvFile = &v
	return this
}

func (this *SeqKitReplaceOptions) KeepUntouch(v bool) *SeqKitReplaceOptions {
	this.inner.KeepUntouch = &v
	return this
}

func (this *SeqKitReplaceOptions) KeepKey(v bool) *SeqKitReplaceOptions {
	this.inner.KeepKey = &v
	return this
}

func (this *SeqKitReplaceOptions) KeyCaptIdx(v int) *SeqKitReplaceOptions {
	this.inner.KeyCaptIdx = &v
	return this
}

func (this *SeqKitReplaceOptions) KeyMissRepl(v string) *SeqKitReplaceOptions {
	this.inner.KeyMissRepl = &v
	return this
}

func Replace(input *api.IDataFrame[string], o *SeqKitReplaceOptions) (*api.IDataFrame[string], error) {
	if o == nil {
		o = &SeqKitReplaceOptions{}
	}
	opts := o.inner
	opts.setDefaults()
	libprepare, err := api.AddParam(libSource("Replace"), "opts", OptionsToString(opts))
	if err != nil {
		return nil, err
	}

	return api.MapPartitions[string, string](input, libprepare)
}

File: bigseqkit-py/bigseqkit/replace.py
from bigseqkit.helper import _setDefault, _libSource, _config, _optionsToString, _parseKargs, SeqKitConfig, IDataFrame


class SeqKitReplaceOptions:

    def __init__(self):
        self.__inner = SeqOptions()

    def config(self, v: SeqKitConfig):
        self.__inner.Config = _config(v)

    def pattern(self, v: str):
        self.__inner.Pattern = v

    def replacement(self, v: str):
        self.__inner.Replacement = v

    def nrWidth(self, v: int):
        self.__inner.NrWidth = v

    def bySeq(self, v: bool):
        self.__inner.BySeq = v

    def ignoreCase(self, v: bool):
        self.__inner.IgnoreCase = v

    def kvFile(self, v: str):
        self.__inner.KvFile = v

    def keepUntouch(self, v: bool):
        self.__inner.KeepUntouch = v

    def keepKey(self, v: bool):
        self.__inner.KeepKey = v

    def keyCaptIdx(self, v: int):
        self.__inner.KeyCaptIdx = v

    def keyMissRepl(self, v: str):
        self.__inner.KeyMissRepl = v

    def _run(self, input: IDataFrame, **kwargs):
        opts = self.__inner
        _parseKargs(opts, kwargs)
        opts.setDefaults()
        libprepare = _libSource("Replace").addParam("opts", _optionsToString(opts))
        return input.mapPartitions(libprepare)

class SeqOptions:

    def __init__(self):
        self.Config = None  # KitConfig
        self.Pattern = None  # string
        self.Replacement = None  # string
        self.NrWidth = None  # int
        self.BySeq = None  # bool
        self.IgnoreCase = None  # bool
        self.KvFile = None  # string
        self.KeepUntouch = None  # bool
        self.KeepKey = None  # bool
        self.KeyCaptIdx = None  # int
        self.KeyMissRepl = None  # string

    def setDefaults(self):
        _setDefault(self, "Config", _config(SeqKitConfig())).setDefaults()
        _setDefault(self, "Pattern", "")
        _setDefault(self, "Replacement", "")
        _setDefault(self, "NrWidth", 1)
        _setDefault(self, "BySeq", False)
        _setDefault(self, "IgnoreCase", False)
        _setDefault(self, "KvFile", "")
        _setDefault(self, "KeepUntouch", False)
        _setDefault(self, "KeepKey", False)
        _setDefault(self, "KeyCaptIdx", 1)
        _setDefault(self, "KeyMissRepl", "")


def replace(input: IDataFrame, o: SeqKitReplaceOptions = None, **kwargs):
    if o is None:
        o = SeqKitReplaceOptions()
    return o._run(input, **kwargs)
